### ISSUE TITLE
segfaults when retrieving keywords from pismo document

### ISSUE THREAD

**Issue by briandoll** [peterc/pismo#1]
Environment:
ruby 1.8.7 (2010-01-10 patchlevel 249) [i686-darwin10.3.1]
pismo (0.6.2)
nokogiri (1.4.2)

irb session showing error:
http://gist.github.com/482550

I get this behavior repeatedly with lots of different URLs.  Very strange...

**Comment by peterc** [comment]
Thanks for reporting this. However, the three techniques you've used are functionally equivalent under the hood.

A number of informal bug reports have been given to me over the last couple of months regarding segfaults. Most were for Pismo 0.6.1 which used the ruby-readability library that was triggering a segfault in Nokogiri 1.4.2 (but not 1.4.1). I wrote an all new replacement for ruby-readability which resolved most of the issue for Nokogiri 1.4.2.

This is a tricky issue to debug as I cannot reproduce it at all. Nokogiri 1.4.2 generally, though, does seem to be a lot more susceptible to segfaults than 1.4.1 ever was. I'm rather hoping a 1.4.3 or 1.5 is released that brings back some stability..

**Comment by kris** [comment]
I can confirm the same issue. Not chaining causes consistent segfaults.

**Comment by kris** [comment]
Now that I look at it, it's completely random. It segfaults pretty much all the time now. The Nokogiri guys don't seem to be much help.

**Comment by briandoll** [comment]
This _may_ be an issue with irb.  I can't explain how this may happen, but I can now make it go away in an irb session.

I've updated the gist: http://gist.github.com/482550

Essentially, if we omit spitting the pismo document back out to the irb screen, we can safely interact with it.  This is that old trick where the results of an expression return a ton of data that you don't need to see, so you just stick a puts statement after it.

Also, If I put this code in a simple ruby script, it all works fine, continuing to indicate that displaying the results in irb is the root issue.

**Comment by peterc** [comment]
Ah, actually, I had a similar complaint over e-mail a few weeks back that had the same resolution.. sorry I didn't remember ;-)

<< We realized that the issue was with irb (v 0.9.5). We tried using Pismo within a script to process those links and we had no issues. And we couldn't reproduce those issues with the 'bin/pismo' script either. So I guess we just need to be more careful if we use Pismo with irb. >>
